lib/improver/tests/argparser/test_ArgParser.py

```diff
@@ -311,6 +314,8 @@ class Test_wrong_args_error(unittest.TestCase):
 
     """Test the wrong_args_error method."""
 
+    @ManageWarnings(
```

Comment:
Discussed trying an extra nested while, to try closing the file properly

Reply:
Done.